staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go

```diff
@@ -177,15 +182,19 @@ func (c *controllerCommon) DetachDisk(diskName, diskURI string, nodeName types.N
 
 	// make the lock here as small as possible
 	diskOpMutex.LockKey(instanceid)
+	c.diskAttachDetachMap.Store(strings.ToLower(diskURI), "")
 	resp, err := vmset.DetachDisk(diskName, diskURI, nodeName)
+	c.diskAttachDetachMap.Delete(strings.ToLower(diskURI))
```

Comment:
What happens if the controller restarted or switched to another leader (once leader election is active) ? Wouldn’t we lose track of the state ?

Reply:
correct, this PR won't cover this, it requires data persistence, which is quite expensive, may not worthwile.